### ISSUE TITLE
Explorer displays error message after stock drupal install and enabling module

### ISSUE THREAD

**Issue by justinlevi** [drupal-graphql/graphql#146]
After doing a pretty bare bones Drupal 8 setup using Lightning via the BLT project, I downloaded and enabled the latest Dev version of the GraphQL module via composer (3.x-dev). I then enabled the the GraphQL module along with all GraphQL submodules and browsed to the explorer page. 

Not sure if this is a known bug or not but I'm seeing a JS error on the first landing screen. 

```
Error: RootSchemaQuery fields must be an object with field names as keys or a function which returns such an object.
    at n (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:5018)
    at m (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:15103)
    at e.t.GraphQLObjectType.e.getFields (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:19224)
    at http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9333
    at o (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9479)
    at Array.reduce (native)
    at new e (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:11864)
    at i (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:10:1968)
    at http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:5:27565
```

![screen shot 2017-06-12 at 4 14 53 pm](https://user-images.githubusercontent.com/1117989/27053141-55b54ef2-4f8a-11e7-8bf3-9598cbd09613.jpg)

I'm very new to the module and GraphQL so I was just trying to get my bearings and start to work through a few basic tests. 

![screen shot 2017-06-12 at 4 16 50 pm](https://user-images.githubusercontent.com/1117989/27053222-97258cda-4f8a-11e7-810a-5fc4c1af8ad7.jpg)

Didn't find anything on the issue queue when doing a search so thought I'd mention it here.

**Comment by pmelab** [comment]
At least it's not a browser issue. Looks like your schema is still empty. I suppose you tried clearing the cache.

Could you try to enable only *GraphQL Core*, then add *GraphQL Content* and then the rest and check between each step?

**Comment by justinlevi** [comment]
After clearing cache I get the following error: 

```
Error: SearchApiTaskSearchApiTaskInput fields must be an object with field names as keys or a function which returns such an object.
    at n (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:5018)
    at e.t.GraphQLInputObjectType.e._defineFieldMap (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:22002)
    at e.t.GraphQLInputObjectType.e.getFields (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:21742)
    at http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9333
    at o (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9479)
    at o (http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:8886)
    at Array.reduce (native)
    at http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9451
    at Array.forEach (native)
    at http://blt.dev/modules/contrib/graphql/js/dist/bundle.min.js?v=8.3.3:7:9360
```

No Errors with the following: 

1.  Disable everything and just enable `GraphQL Core`
2.  Enable `GraphQL Content`. 
3. Enable everything else except the `GraphQL Mutations` and the `GraphQL Content Mutations`

Error shows up: 
4. Enable the `GraphQL Mutations` and `GraphQL Content Mutations`

The Error goes away if I disable the `GraphQL Content Mutations` again. I don't yet know what that module actually does for me so not sure if that error is just my lack of understanding or if it's a bug. 

From a very high level perspective though a module probably shouldn't present any error after being enabled.

**Comment by fubhy** [comment]
Yes, this is definitely not right. And your last comment also clarifies the issue for me and I know what I need to do to fix it now. Thanks for reporting back!

Explanation: The input field deriver for input types on content entities can yield 0 fields if there are simply no editable fields on a given content entity. In that case, the input type should not end up in the schema either.

I will come up with a fix for that.

Thanks @justinlevi!

**Comment by fubhy** [comment]
Oh and regarding your question for the "mutation" modules.

"Mutations" are queries that create/update/delete stuff. Mutations can be used to e.g. create an entity via GraphQL and then return the created entity as output in your GraphQL request. If you don't need mutations, you can just leave those modules disabled for now. In general you don't need to enable all modules anyways, just the things that give you the features for the schema that you need. Once you get rid of the error in /graphql/explorer you will be able to play around with the schema and write some test queries :).

**Comment by fubhy** [comment]
This will be fixed through #150
